app.py

```diff
@@ -87,7 +87,9 @@ def gen(camera):
             socketio.emit('gameover', {'data': 'gameover'})
         if camera.get_game().check_outpipe():
             socketio.emit('out_pipe', {'data': 'out_pipe'})
-            print("out_pipe")
+        if camera.get_game().check_heart_reset():
+            socketio.emit('heart_reset', {'data': 'heart_reset'})
+            print("heart reset.")
```

Comment:
Only `battle game` owns this variable.
This function may not need, it can do in frontend.
Check the heart number while websocket received, then redraw the heart.
TBD.

Reply:
此題無解~~

Reply:
Add a FIXME comment for this problem then merge this PR.